.github/workflows/dockerhub.yml

```
@@ -0,0 +1,25 @@
+name: Push to DockerHub
+
+on:
+  push:
+    branches: [master]
+
+jobs:
+  dockerhub:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - uses: satackey/action-docker-layer-caching@v0.0.5
+
+      - name: Build and push Docker images
+        uses: docker/build-push-action@v1.1.0
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_PASSWORD }}
+          repository: ${{ github.event.repository.full_name }}
+          tag_with_sha: true
+          tags: latest
+          dockerfile: Dockerfile
```

Looks like we should upgrade to v2, but we can do that in a later PR - it seems like a big release that requires significant changes. See the [upgrade notes](https://github.com/docker/build-push-action/blob/master/UPGRADE.md#simple-workflow).

In particular, the tagging part seems much more involved in v2.

Don't accept the below suggestion - it won't work. But I think it's close.

```suggestion
      - name: Docker meta
        id: docker_meta
        uses: crazy-max/ghaction-docker-meta@v1
        with:
          images: foo/bar # ?
          
      - name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v1

      - name: Login to DockerHub
        uses: docker/login-action@v1
        with:
          username: ${{ secrets.DOCKER_USERNAME }}
          password: ${{ secrets.DOCKER_PASSWORD }}

      - name: Build and push Docker images      
        uses: docker/build-push-action@v2.1.0
        with:
          file: ./Dockerfile
          tags: ${{ steps.docker_meta.outputs.tags }}
          labels: ${{ steps.docker_meta.outputs.labels }}
```

---

Interesting. I didn't know they changed it.

---

v2 was only released 2 weeks ago. In the latest v1 they've added a deprecation notice that advises upgrading to v2, but that wasn't included in the v1.1.0 we were using. I'll make a suggestion that just pins to the v1.1.2 hash for this PR.